File: api/Reactello.Domain/Entities/Note.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reactello.Domain.Entities
{
    public class Note : EntityBase
    {
        public Note()
        {
            Comments = new HashSet<Comment>();
        }
        public string Title { get; set; }
        public string Description { get; set; }
        public int Position { get; set; }
        public ICollection<Comment> Comments { get; private set; }

        public Section Section { get; set; }
    }
}

File: src/models/note.ts
import { Comment } from './comment';
import { NormalizedModel, NormalizedNote } from './normalizedModels';

export interface NoteCreate {
	title: string;
	description: string;
	sectionId: number;
}
export interface Note {
	id: number;
	title: string;
	description: string;
	position: number;
	sectionId: number;
	comments: Comment[];
}

export const normalize = (note: Note): NormalizedModel<NormalizedNote> => {
	return {
		[note.id]: {
			id: note.id,
			title: note.title,
			position: note.position,
			description: note.description,
			sectionId: note.sectionId,
			comments: note.comments.map((x) => x.id)
		}
	};
};
